support-workers/src/main/scala/com/gu/emailservices/DigitalPackEmailFields.scala

```diff
@@ -1,78 +1,242 @@
 package com.gu.emailservices
 
+import cats.implicits._
+import com.gu.emailservices.DigitalSubscriptionEmailAttributes.PaymentFieldsAttributes
+import com.gu.emailservices.DigitalSubscriptionEmailAttributes.PaymentFieldsAttributes.{CCAttributes, DDAttributes, PPAttributes}
 import com.gu.emailservices.SubscriptionEmailFieldHelpers._
-import com.gu.i18n.Currency
-import com.gu.support.promotions.Promotion
 import com.gu.support.workers._
 import com.gu.support.workers.states.PaymentMethodWithSchedule
+import com.gu.support.zuora.api.ReaderType
+import io.circe._
+import io.circe.generic.semiauto.deriveEncoder
+import io.circe.syntax._
 
-object DigitalPackEmailFields {
+sealed trait DigitalSubscriptionEmailAttributes extends Product with Serializable
+object DigitalSubscriptionEmailAttributes {
+
+  implicit val e1: Encoder.AsObject[BasicDSAttributes] = deriveEncoder
+  implicit val e2: Encoder.AsObject[DirectDSAttributes] = deriveEncoder
+  implicit val e3: Encoder.AsObject[GifteeRedemptionAttributes] = deriveEncoder
+  implicit val e4: Encoder.AsObject[GifterPurchaseAttributes] = deriveEncoder
+  implicit val e5: Encoder.AsObject[GifteeNotificationAttributes] = deriveEncoder
+
+  implicit val encoder: Encoder.AsObject[DigitalSubscriptionEmailAttributes] = Encoder.AsObject.instance {
+    case v: BasicDSAttributes => v.asJsonObject
```

Comment:
Why encode to JsonObject rather than Json?

Reply:
more specific type, a Json could also be a JArray, JString, JNumber etc, so it saves checking that before we start traversing in the flattening function.

I'm not totally sold on it (as I prefer json stuff to be as totally standard as possible, as anything new always requires consuilting the docs) so happy to change back and have the extra bit of scala code and a runtime check if you prefer?

Reply:
No happy to go with this